packages/frontend/src/context/positions.tsx

```diff
@@ -0,0 +1,264 @@
+          acc.totalETHSpent = acc.totalETHSpent.plus(wethAmt)
+          acc.totalUSDSpent = acc.totalUSDSpent.plus(usdAmt)
+
+          acc.usdAmount = acc.usdAmount.plus(usdAmt)
+          if (acc.squeethAmount.isZero()) {
+            acc.usdAmount = bigZero
+            acc.wethAmount = bigZero
+            acc.totalSqueeth = bigZero
+            acc.totalETHSpent = bigZero
+            acc.totalUSDSpent = bigZero
+          }
+
+          return acc
+        },
+        {
+          squeethAmount: bigZero,
+          wethAmount: bigZero,
+          usdAmount: bigZero,
+          totalSqueeth: bigZero,
+          totalETHSpent: bigZero,
+          totalUSDSpent: bigZero,
+        },
+      ) || {
+        squeethAmount: bigZero,
+        wethAmount: bigZero,
+        usdAmount: bigZero,
+        totalSqueeth: bigZero,
+        totalETHSpent: bigZero,
+        totalUSDSpent: bigZero,
+      },
+    [isWethToken0, swaps?.length],
+  )
+
+  const mintedDebt = useMemo(() => {
+    // squeethAmount = user long balance if oSqueethBal > 0, but it could also be minted balance
+    return shortVaults[firstValidVault]?.shortAmount.gt(0) &&
+      oSqueethBal?.isGreaterThan(0) &&
+      positionType === PositionType.LONG
+      ? oSqueethBal.minus(squeethAmount)
+      : shortVaults[firstValidVault]?.shortAmount.gt(0) && oSqueethBal?.isGreaterThan(0)
+      ? oSqueethBal
+      : new BigNumber(0)
+  }, [firstValidVault, oSqueethBal.toString(), positionType, shortVaults?.length, squeethAmount.toString()])
+
+  const shortDebt = useMemo(() => {
+    return positionType === PositionType.SHORT ? squeethAmount : new BigNumber(0)
+  }, [positionType, squeethAmount.toString()])
+
+  const longSqthBal = useMemo(() => {
+    return mintedDebt.gt(0) ? oSqueethBal.minus(mintedDebt) : oSqueethBal
+  }, [oSqueethBal.toString(), mintedDebt.toString()])
+
+  const lpDebt = useMemo(() => {
+    return depositedSqueeth.minus(withdrawnSqueeth).isGreaterThan(0)
+      ? depositedSqueeth.minus(withdrawnSqueeth)
+      : new BigNumber(0)
+  }, [depositedSqueeth.toString(), withdrawnSqueeth.toString()])
+
+  const { finalSqueeth, finalWeth } = useMemo(() => {
+    // dont include LPed & minted amount will be the correct short amount
+    const finalSqueeth = squeethAmount
+    const finalWeth = wethAmount.div(squeethAmount).multipliedBy(finalSqueeth)
+    return { finalSqueeth, finalWeth }
+  }, [squeethAmount.toString(), wethAmount.toString()])
+
+  useEffect(() => {
+    if (finalSqueeth.isGreaterThan(0)) {
+      setPositionType(PositionType.LONG)
+    } else if (finalSqueeth.isLessThan(0)) {
+      setPositionType(PositionType.SHORT)
+    } else setPositionType(PositionType.NONE)
+  }, [finalSqueeth.toString()])
+
+  useEffect(() => {
+    for (let i = 0; i < shortVaults.length; i++) {
+      if (shortVaults[i]?.collateralAmount.isGreaterThan(0)) {
+        setFirstValidVault(i)
+      }
+    }
+  }, [shortVaults.length])
+
+  useEffect(() => {
```

Comment:
Thinking that we could remove 217-236, and move existingCollatR & existingLiquidationPrice & existingCollat & first valid vault states to useVaultData hook(or maybe we could create 1 context for positions and separate the swaps cals & vault states & LP position calcs in three hooks and then combine the results in position context and then calculate the minted/LPed/shortDebt)

Reply:
Knowing that they are all related to positions, but not sure if it's a best practice that we put all the position calcs and data pulling in one context

Reply:
sgtm! curious @KMKoushik thoughts as well?